disnake/state.py

```diff
@@ -1262,17 +1269,19 @@ def parse_guild_member_update(self, data) -> None:
         if member is not None:
             old_member = Member._copy(member)
             member._update(data)
-            user_update = member._update_inner_user(user)
+            user_update = member._update_inner_user(data["user"])
             if user_update:
                 self.dispatch("user_update", user_update[0], user_update[1])
 
             self.dispatch("member_update", old_member, member)
+            user = member
         else:
+            user = self.store_user(data["user"])
```

Comment:
The user should only be deserialized if necessary, something like this should avoid reading data twice:
```diff
diff --git a/disnake/state.py b/disnake/state.py
index bde5c1ef..000ee1e5 100644
--- a/disnake/state.py
+++ b/disnake/state.py
@@ -1276,7 +1276,6 @@ class ConnectionState:
             self.dispatch("member_update", old_member, member)
             user = member
         else:
-            user = self.store_user(data["user"])
             if self.member_cache_flags.joined:
                 member = Member(data=data, guild=guild, state=self)
 
@@ -1286,6 +1285,9 @@ class ConnectionState:
                     self.dispatch("user_update", user_update[0], user_update[1])
 
                 guild._add_member(member)
+                user = member
+            else:
+                user = self.store_user(data["user"])
             _log.debug(
                 "GUILD_MEMBER_UPDATE referencing an unknown member ID: %s. Discarding.", user_id
             )

```

Reply:
53365f9c